core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -180,4 +180,21 @@ public byte[] toArray() {
   public String toString() {
     return new String(data, offset, length, UTF_8);
   }
+
+  public void setArray(byte[] data, int offset, int length) {
+    if (offset < 0 || offset > data.length || length < 0 || (offset + length) > data.length) {
+      throw new IllegalArgumentException(" Bad offset and/or length data.length = " + data.length
+          + " offset = " + offset + " length = " + length);
+    }
+    this.data = data;
+    this.offset = offset;
+    this.length = length;
+  }
+
+  public void setLength(int length) {
+    if (length < 0) {
+      throw new IllegalArgumentException(" Bad length = " + length);
+    }
+    this.length = length;
+  }
```

Comment:
Wondering if we can drop this method, not sure.   Its only called in two places, wonder if it would be cleaner to refactor those two places to call `setArray` instead of this setLength method.  

If this method is kept then its validation will need to be improved to ensure that length makes sense for the current offset and data.length.   Also need to add javadoc that states it sets the length if the length is less than the length of the current array minus the current offset.  Also the javadoc would need to include a `@since 3.1.0` tag.

Reply:
Agreed that it can be dropped. Updated.